langchain/vectorstores/azuresearch.py

```diff
@@ -209,17 +211,23 @@ def add_texts(
             key = base64.urlsafe_b64encode(bytes(key, "utf-8")).decode("ascii")
             metadata = metadatas[i] if metadatas else {}
             # Add data to index
-            data.append(
-                {
-                    "@search.action": "upload",
-                    FIELDS_ID: key,
-                    FIELDS_CONTENT: text,
-                    FIELDS_CONTENT_VECTOR: np.array(
-                        self.embedding_function(text), dtype=np.float32
-                    ).tolist(),
-                    FIELDS_METADATA: json.dumps(metadata),
-                }
-            )
+            data_item = {
+                "@search.action": "upload",
+                FIELDS_ID: key,
+                FIELDS_CONTENT: text,
+                FIELDS_CONTENT_VECTOR: np.array(
+                    self.embedding_function(text), dtype=np.float32
+                ).tolist(),
+                FIELDS_METADATA: json.dumps(metadata),
+            }
+            if extra_fields:
+                # Add the extra fields to the data item dictionary
+                for field_list in extra_fields:
+                    if len(field_list) > i:
```

Comment:
is there any reason NOT to treat this the same as metadata (aka assume its the same length as texts and do `_extra_fields = extra_fields[i] if extra_fields else {}`

imo we should

Reply:
The metadata is dumped as json now and will be treated as a string when extracted. eg : "metadata": "{"document_id": "xcf15"}"
This field is retrievable but not searchable or filterable. As we may need to pass in a filter like

_filters = {
"document_id eq 'id_01'"
}

retriever = AzureSearchVectorStoreRetriever(
vectorstore = vector_store,
filters=filters,
k = 4)_

extra_fields is an optional dictionary that defaults to 'None'. We will be able to apply filters if a dictionary is passed to add texts function like
extra_fields = [[{'document_id': 'xcf0'},{'document_id': 'xcf1'}],[{'document_id': 'xcf2'},{'document_id': 'xcf3'}]]
vector_store.add_texts(texts, metadatas, extra_fields)

Reply:
Thanks..It was a great suggestion
I have created bc9251f32be24aea8042f32b5741e3cc91412703 for this.